java/src/main/java/com/logicalclocks/hsfs/engine/hudi/HudiEngine.java

```diff
@@ -134,6 +142,12 @@ public void saveHudiFeatureGroup(SparkSession sparkSession, FeatureGroupBase fea
 
     Map<String, String> hudiArgs = setupHudiWriteOpts(featureGroup, operation, writeOptions);
 
+    for (Map.Entry<String, String> entry : HUDI_DEFAULT_PARALLELISM.entrySet()) {
+      if (writeOptions != null && !writeOptions.containsKey((entry.getKey()))) {
+        hudiArgs.put(entry.getKey(), entry.getValue());
+      }
+    }
```

Comment:
We have a method called `setHudiWriteOptions` why don't we set those options there? Sot that they are reused across (e.g. by the delete method as well) and not just by the save method.

Also as for the other Hudi options, you can just set them and let the user overwrite them with the `putAll` call we have here - https://github.com/logicalclocks/feature-store-api/blob/1a421731b0c45af0e4f1a02e96ecfcb93fa10488/java/src/main/java/com/logicalclocks/hsfs/engine/hudi/HudiEngine.java#L249

Reply:
regarding `hudiArgs.putAll(writeOptions)` my idea was if they provide some option we don't override them.

Reply:
Yes you are correct that we should let the user overwrite the options. My argument is that you can just add the following code
```
 for (Map.Entry<String, String> entry : HUDI_DEFAULT_PARALLELISM.entrySet()) {
          hudiArgs.put(entry.getKey(), entry.getValue());
      }
```
and then if the user wants to overwrite them, the `putAll` will take care of setting the user preference, but I believe what I'm suggesting is a cleaner way of writing the code.